src/teaching-element/Assessment/MultipleChoice.vue

```diff
@@ -36,22 +37,32 @@ export default {
     submission: { type: Array, default: () => ([]) }
   },
   data() {
-    return { userAnswer: this.submission || [] };
+    return {
+      userAnswer: this.submission || [],
+      unorderedAnswer: this.submission || []
+    };
   },
   computed: {
     type() {
       const options = this.options.multipleChoice || defaults;
       return options.type;
+    },
+    randomAnswers() {
+      return shuffle(this.answers);
     }
   },
   methods: {
     update() {
+      this.userAnswer = [];
+      this.unorderedAnswer.forEach(a => {
```

Comment:
This logic can be a computed meaning in this assessment the userAnswer should be a computed.